### PR TITLE
DS-3030: Add more names to statistic.list

### PR DESCRIPTION
**Pull request by chschan**

### CODE REVIEW

File: R/ParseUserEnteredTable.R [hunk 1]
```diff
@@ -160,7 +160,10 @@ parseAsVectorOrMatrix <- function(m, warn = FALSE)
         return(out)
     }
 
-    statistic.list <- c("", "%", "Column %", "Row %", "Total %")
+    statistic.list <- c("", "%", "Column %", "Row %", "Total %",
```
Comment (mwmclean) [inline R/ParseUserEnteredTable.R]: Is this really the entire list of Q table statistics?

File: R/ParseUserEnteredTable.R [hunk 2]
```diff
@@ -160,7 +160,10 @@ parseAsVectorOrMatrix <- function(m, warn = FALSE)
...
+        "% Column Share", "% Row Share", "% Total Share",
+        "% Responses", "% Row Responses", "% Column Responses", "% Total Responses",
+        "Cumulative %", "Expected %", "Residual %")
     # These are not used be cause they can plausibly be row/column labels
     #   "n", "Average", "Standard Error", "Population")
```
Comment (mwmclean) [inline R/ParseUserEnteredTable.R]: How much of a danger is it really supporting these extra statistics given we're only checking the 1,1 entry?

File: R/ParseUserEnteredTable.R [hunk 3]
```diff
@@ -160,51 +160,96 @@ parseAsVectorOrMatrix <- function(m, warn = FALSE)
...
-    idx <- if (first.entry.chars) -1
-           else seq_len(n.col)
-    col.names.given <- !isTextNumeric(m[1, idx], allow.missing = TRUE)
-    idx <- if (first.entry.chars) -1
-           else seq_len(n.row)
-    if (col.names.given)
+    row.names.given <- isPossibleLabel(m[-1,1])
+    col.names.given <- isPossibleLabel(m[1,-1])
+
+    statistic.list <- c("", "%", "% Column Responses", "% Column Share",
```
Comment (mwmclean) [inline R/ParseUserEnteredTable.R]: Please move this to an internal function `getQStatisticList()` or similar that simply returns this character vector.

File: R/ParseUserEnteredTable.R [hunk 4]
```diff
@@ -226,3 +271,23 @@ parseAsVectorOrMatrix <- function(m, warn = FALSE)
     }
     out
 }
+
+# Returns a logical indicating whether the vector of text might be a label
+# Returns NA if they are all whole integers or can be parsed as a date
+isPossibleLabel <- function(t)
+{
+    v <- asNumericVector(t)
+    if (any(isMissing(t)) && any(!is.na(v)) && all(!is.na(v) | isMissing(t)))
```
Comment (mwmclean) [inline R/ParseUserEnteredTable.R]: What is this checking? It looks weird with the `&&`s. Assign the expression inside the `if` to a variable with an informative name please.

File: tests/testthat/test-ParseUserEnteredTable.R [hunk 5]
```diff
@@ -406,3 +406,85 @@ test_that("Row and Column titles",
...
+    raw <- structure(c("Random statistic", "Other", "Burger Shack", "Nuovo Burger",
+        "Arnold's", "Ma's burgers", "Burger Chef", "Apr-Jun 15", "60",
+        "2", "0", "21", "2", "11"), .Dim = c(7L, 2L))
+    res <- ParseEnteredData(raw)
+    expect_equal(res, structure(c(60, 2, 0, 21, 2, 11),
+        .Dim = c(6L, 1L), .Dimnames = list(
+        c("Other", "Burger Shack", "Nuovo Burger", "Arnold's", "Ma's burgers",
+        "Burger Chef"), "Apr-Jun 15"), statistic = "Random statistic"))
+
+    # NA + NA + match statistic
```
Comment (mwmclean) [inline tests/testthat/test-ParseUserEnteredTable.R]: What does NA, F, and T mean here?

File: R/ParseUserEnteredTable.R (hunk 2 above)
Comment (chschan) [reply R/ParseUserEnteredTable.R]: I think its ok to support the extra statistics and they are now included in `isQStatistic`

File: R/ParseUserEnteredTable.R (hunk 4 above)
Comment (chschan) [reply R/ParseUserEnteredTable.R]: In 40b420c, I've assigned the expression to `has.numeric.and.missing`

File: R/ParseUserEnteredTable.R (hunk 3 above)
Comment (chschan) [reply R/ParseUserEnteredTable.R]: moved to isQStatistic in 40b420c

File: tests/testthat/test-ParseUserEnteredTable.R (hunk 5 above)
Comment (chschan) [reply tests/testthat/test-ParseUserEnteredTable.R]: Here we through the combinations of whether the 1st row/column is or may possible be a vector of labels, and whether the 1,1, entry matches a statistic. Row and column labels are treated in the same way, so they are interchangeable. I have fixed up the comments in 40b420c so they make more sense.

**Pull request merged**